advanced/Scripts/piholeDebug.sh

```diff
@@ -653,8 +621,7 @@ detect_ip_addresses() {
         log_write "${TICK} IPv${protocol} address(es) bound to the ${PIHOLE_INTERFACE} interface:"
         # Since there may be more than one IP address, store them in an array
         for i in "${!ip_addr_list[@]}"; do
-            # For each one in the list, print it out
-            does_ip_match_setup_vars "${protocol}" "${ip_addr_list[$i]}"
+            log_write "    ${ip_addr_list[$i]}"
```

Comment:
Is there any calls to `pihole-FTL` that will expose what IP addresses it will respond to? Can we add them here if there is that functionality?

Reply:
No, this does not exist. `dnsmasq` determines this at run-time. We could rebuild this for ourselves but it would be quite involved.

Reply:
Okay, might be something to put on the list to consider.

Would be nice to have `pihole-FTL --list-address` or similar that would output a list of the interface and the address it is using for that interface.

Reply:
I guess one has to deduct from `PIHOLE_INTERFACE=XXXX` and the listening behavior based on `/etc/dnsmasq.d/01-pihole.conf` and `Name resolution (IPv4/IPv6) using a random blocked domain and a known ad-serving domain`?

Reply:
The main issue I see right now is that it would launch a dedicated engine (instead of interfacing with the existing one). So the new engine would need to read and parse the config and find out itself how the currently running engine *might* work (if the config wasn't changed).

We may rather add an API endpoint for this to query this. We'd need a JSON deserializer for bash here, but that shouldn't be an issue (PADD / chronometer needs this as well as dependency).

Reply:
`jq`

Reply:
@yubiuser I didn't see your reply. Mind there may be additional lines in any user-provided config files that may contain lines like `listen-address=...`

Reply:
Would we be able to set a CHAOS or TXT record that emits that information when queried?

Reply:
Yes, that would maybe possible. `dnsmasq` has a rather complex determination if it should listen to an address. There are various modes, the default being listen on wildcards and possibly ignore some when we should not listen to them. This can only be determined at runtime and I wouldn't know how to, e.g., specify "listen to all but a few subnets" as CHAOS string. Encoding and parsing would be quite a complex task.

It might be easier to ask for a specific address and let `dnsmasq` check if it would listen to this one. But is this what we need?